R/ci.R

```diff
@@ -1,82 +1,104 @@
+      #'    (overwrite after confirmation in interactive mode only)
+      travis:::use_travis_yml() #FIXME: Export function in travis
+    }
 
-  travis_activate(browse)
-  use_travis_badge()
+    repo_type <- travis:::detect_repo_type() #FIXME: Export function in travis
 
-  invisible(TRUE)
-}
+    if ("Appveyor" %in% services) {
+      #' 1. Create a default `appveyor.yml` file
+      #'    (depending on repo type, overwrite after confirmation
+      #'    in interactive mode only)
+      if (travis:::needs_appveyor(repo_type)) travis:::use_appveyor_yml() #FIXME: Export function in travis
+    }
 
-use_travis_badge <- function() {
-  check_uses_github()
+    #' 1. Create a default `tic.R` file depending on the repo type
+    #'    (package, website, bookdown, ...)
+    use_tic(repo_type)
 
-  url <- glue("https://travis-ci.org/{github_repo_spec()}")
-  img <- glue("{url}.svg?branch=master")
+    #' 1. Enable deployment (if necessary, depending on repo type)
+    #'    via [use_travis_deploy()]
+    if (travis:::needs_deploy(repo_type)) use_travis_deploy() #FIXME: Export function in travis
 
-  use_badge("Travis build status", url, img)
+    #' 1. Create a GitHub PAT and install it on Travis CI via [travis_set_pat()]
+    travis_set_pat()
+  })
 }
 
-travis_activate <- function(browse = interactive()) {
-  url <- glue("https://travis-ci.org/profile/{github_owner()}")
-
-  todo("Turn on travis for your repo at {url}")
-  if (browse) {
-    utils::browseURL(url)
-  }
+use_tic <- function(repo_type) {
+  use_template_tic(repo_type, "tic.R")
 }
 
-uses_travis <- function(base_path = proj_get()) {
-  path <- glue("{base_path}/.travis.yml")
-  file_exists(path)
+use_template_tic <- function(..., target = basename(file.path(...))) {
+  source <- template_file(...)
+  safe_filecopy(source, target)
+  message("Added ", target, " from template.")
+  use_build_ignore(target)
 }
 
-check_uses_travis <- function(base_path = proj_get()) {
-  if (uses_travis(base_path)) {
-    return(invisible())
-  }
+use_travis_badge <- function() {
+  uses_github()
 
-  stop_glue(
-    "Cannot detect that package {value(project_name(base_path))}",
-    " already uses Travis.\n",
-    "Do you need to run {code('use_travis()')}?"
-  )
+  url <- glue("https://travis-ci.org/{github_repo_spec()}")
+  img <- glue("{url}.svg?branch=master")
+
+  use_badge("Travis build status", url, img)
 }
 
+#' Add test coverage via codecov or coveralls
+#'
+#' Enables test coverage reporting via codecov or coveralls
+#'
+#' @importFrom travis travis_is_enabled
 #' @section `use_coverage()`:
 #' Adds test coverage reports to a package that is already using Travis CI.
-#' @rdname ci
+#' @name use_coverage
 #' @param type Which web service to use for test reporting. Currently supports
 #'   [Codecov](https://codecov.io) and [Coveralls](https://coveralls.io).
 #' @export
 use_coverage <- function(type = c("codecov", "coveralls")) {
-  check_uses_travis()
+  travis_is_enabled()
```

Comment:
This function doesn't have side effects.

The old implementation looked at the presence of a `.travis.yml` file, now we contact the API. I'm not sure which is better. Also, this function is still suggesting to edit `.travis.yml` .

Reply:
So which to keep? I thought using all the travis related stuff from `travis` is more tidy than having some "travis_*" functions in 'usethis`.

Reply:
We still can have a utility wrapper in _usethis_: `check_uses_travis()` could call `travis_is_enabled()` and stop if not.

Reply:
I wrapped `travis_is_enabled()` into `check_uses_travis()`.